Go/src/mqsamputils/connection.go

```diff
@@ -97,9 +97,7 @@ func CreateConnection(index int) (ibmmq.MQQueueManager, error) {
 			return ibmmq.MQQueueManager{}, err
 		}
 
-	}
-
-	if username := env.User; username != "" {
+	} else if username := env.User; username != "" {
```

Comment:
As a whole the function `CreateConnection` is approaching 100 lines in length. Is there a way to refactor some of the sections into separate functions?